### PR TITLE
chore: Add descriptions to all component props

### PR DESCRIPTION
**Pull request by VincentSmedinga**
These descriptions will show up in Storybook and IDEs. I tried to keep them very compact.

### CODE REVIEW

**Comment by RubenSibon** [comment]
That's a lot of new text. How shall we review this? Divide in parts?

**Comment by VincentSmedinga** [comment]
> That's a lot of new text. How shall we review this? Divide in parts?

It’s just static text, some patterns and even identical instances between them. I think you can do it 💪🏼

File: storybook/src/components/IconButton/IconButton.docs.mdx
Comment (dlnr) [inline storybook/src/components/IconButton/IconButton.docs.mdx:1]: Perhaps we should describe (here or in the upcomming colour docs) that the [contrast ratio for non-text should be 3:1](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html)

**Review by dlnr — APPROVED** [review]
Only one suggestion but good work

File: storybook/src/components/IconButton/IconButton.docs.mdx
Comment (VincentSmedinga) [reply storybook/src/components/IconButton/IconButton.docs.mdx:1]: Yes, added this to the appropriate ticket on our backlog.

File: storybook/src/components/TextArea/TextArea.stories.tsx [hunk 1]
```diff
@@ -20,22 +20,18 @@ const meta = {
...
     },
-    cols: {
-      control: {
-        type: 'number',
-      },
-    },
-    rows: {
-      control: {
-        type: 'number',
-      },
```
Comment (alimpens) [inline storybook/src/components/TextArea/TextArea.stories.tsx:38]: Why did you remove these? They show functionality we built into TextArea.

File: packages/react/src/TopTaskLink/TopTaskLink.tsx [hunk 2]
```diff
@@ -9,7 +9,9 @@ import type { AnchorHTMLAttributes, ForwardedRef } from 'react'
 import { VisuallyHidden } from '../VisuallyHidden'
 
 export type TopTaskLinkProps = {
+  /** The heading. */
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages/react/src/TopTaskLink/TopTaskLink.tsx:12]: Not a Heading component no, but why mix both terms? we chose the word ’heading’ for titles.

File: storybook/src/components/TextArea/TextArea.stories.tsx (hunk 1 above)
Comment (VincentSmedinga) [reply storybook/src/components/TextArea/TextArea.stories.tsx]: True, I’ve restored them.

File: storybook/src/components/Select/Select.stories.tsx [hunk 3]
```diff
@@ -36,6 +36,14 @@ const meta = {
       </Select.Option>,
     ],
   },
+  argTypes: {
+    disabled: {
+      description: 'Prevents interaction. Use sparingly.',
+    },
+    invalid: {
+      description: 'Whether the value fails a validation rule.',
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/components/Select/Select.stories.tsx:44]: Correct

File: storybook/src/components/SearchField/SearchField.stories.tsx [hunk 4]
```diff
@@ -24,15 +25,13 @@ const meta = {
   },
   argTypes: {
     label: {
-      control: 'text',
+      description: 'Describes the field. Not displayed visually.',
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/components/SearchField/SearchField.stories.tsx:28]: No, this one wouldn’t show up if we removed this because the prop is on a subcomponent.

File: storybook/src/components/Link/Link.stories.tsx [hunk 5]
```diff
@@ -28,28 +29,14 @@ const meta = {
...
-      },
-    },
-    target: {
-      name: 'target',
-      type: { name: 'string', required: false },
-      table: {
-        category: 'HTML attribute',
-        defaultValue: { summary: '' },
-      },
     },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/components/Link/Link.stories.tsx:40]: Do you mean `rel`? We do mention it in the guidelines.

File: packages/react/src/Select/Select.tsx [hunk 6]
```diff
@@ -10,7 +10,10 @@ import { SelectOption } from './SelectOption'
 import { SelectOptionGroup } from './SelectOptionGroup'
 
 export type SelectProps = {
-  /** There is no native invalid attribute for select, but you can use this to get the same result as other form components */
+  /**
+   * Whether the value fails a validation rule.
+   * A native `invalid` attribute does not exist for `select`, but you can use this to get the same result as for other form components.
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages/react/src/Select/Select.tsx:15]: I didn’t understand either; it was already there. Not anymore.

File: packages/react/src/Breadcrumb/BreadcrumbItem.tsx [hunk 7]
```diff
@@ -8,6 +8,7 @@ import { forwardRef } from 'react'
 import type { ForwardedRef, HTMLAttributes, PropsWithChildren } from 'react'
 
 export type BreadcrumbItemProps = {
+  /** The url for the link. */
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages/react/src/Breadcrumb/BreadcrumbItem.tsx]: We removed this explicit field from the type in #1232.

File: packages/react/src/Select/Select.tsx [hunk 8]
```diff
@@ -10,7 +10,9 @@ import { SelectOption } from './SelectOption'
 import { SelectOptionGroup } from './SelectOptionGroup'
 
 export type SelectProps = {
-  /** There is no native invalid attribute for select, but you can use this to get the same result as other form components */
+  /**
+   * Whether the value fails a validation rule.
+   */
```
Comment (alimpens) [inline packages/react/src/Select/Select.tsx:15]:
```suggestion
  /** Whether the value fails a validation rule. */
```

File: packages/react/src/TopTaskLink/TopTaskLink.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/react/src/TopTaskLink/TopTaskLink.tsx:12]: No, we chose the word 'heading' for headings ;) This doesn't render an HTML heading element, so using the word 'heading' is a little confusing imo. 

We use 'title' in the Top Task Link docs consistently as well.

File: storybook/src/components/Link/Link.stories.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (alimpens) [reply storybook/src/components/Link/Link.stories.tsx:40]: Sure, but it's a native attribute which doesn't have any effect on the appearance of the component. Doesn't feel very useful to have it as a control imo.

File: storybook/src/components/SearchField/SearchField.stories.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (alimpens) [reply storybook/src/components/SearchField/SearchField.stories.tsx]: Ah, that makes sense

File: storybook/src/components/TextArea/TextArea.stories.tsx [hunk 9]
```diff
@@ -20,22 +20,30 @@ const meta = {
     disabled: false,
   },
   argTypes: {
+    cols: {
+      control: {
+        type: 'number',
+      },
+      description: 'The maximum number of characters per line.',
```
Comment (alimpens) [inline storybook/src/components/TextArea/TextArea.stories.tsx:27]: This description doesn't read to me as if it's about the width of the Text Area.

How about:

```suggestion
      description: 'The width in avarage character widths.',
```

Comment (VincentSmedinga) [reply storybook/src/components/TextArea/TextArea.stories.tsx]: True. I’ve made it ‘The width, expressed in the average number of characters’.

File: documentation/storybook.md [hunk 10]
```diff
@@ -21,18 +21,36 @@ We write our documentation in English, the stories are Dutch.
 ## Best practices for controls
 
 Controls are automatically generated based on the component’s typing.
-If you want to document native HTML attributes, you can use [`argTypes`](https://storybook.js.org/docs/api/arg-types).
-You can also use `argTypes` to override the automatically generated controls.
+
+### Prop descriptions
+
+We use JSDoc to describe each prop to keep documentation close to the actual definition.
+This also helps IDE’s pick up the description and display it in their user interface.
```
Comment (alimpens) [inline documentation/storybook.md:28]:
```suggestion
This also helps IDEs pick up the description and display it in their user interface.
```

**Review by alimpens — APPROVED** [review]
